homeassistant/components/opentherm_gw/climate.py

```diff
@@ -39,6 +39,7 @@ def __init__(self, hass, config):
         self.temp_precision = config.get(CONF_PRECISION)
         self._current_operation = STATE_IDLE
         self._current_temperature = None
+        self._new_target_temperature = None
```

Comment:
I don't see the point of this variable. It seems like we would be good with just `self._target_temperature`, which is set both by the service method and the callback.

Reply:
It may take some time for the new setpoint to propagate to the thermostat. In the mean time, we will keep receiving reports from the gateway with the old setpoint. We use `self._new_target_temperature` to prevent a situation where you change the target temperature in the UI, just to see it jump back to the old value right after until it is propagated to the thermostat. We keep the old value from the thermostat in `self._target_temperature` and use `self._new_target temperature` instead until the next change from the thermostat which should be the new value, but may be something else e.g. in case of manual action on the thermostat.

Reply:
Thanks for the explanation! :+1:

Reply:
Exactly, it was nice to discover this solution by mvn23. Whole thread just FYI. https://github.com/mvn23/pyotgw/issues/6